example/helloWorldLambda/src/helloWorldLambda.cpp

```diff
@@ -150,13 +153,23 @@ auto main()
     // to wrap allready existing code into a
     // std::function and provide it to the alpaka 
     // library.
+#if !BOOST_LANG_CUDA && !BOOST_LANG_HIP
     alpaka::kernel::exec<Acc> (
         queue,
         workDiv,
         std::function<void(Acc&, size_t)>( hiWorldFunction<Acc> ),
         nExclamationMarks);
+#endif
+#if BOOST_LANG_CUDA
```

Comment:
We should not start to use #ifdef within the example because the goal of alpaka is to avoid that. (if this is only for testing all is fine)
We can ship a meta::function with alpaka which is doing `ifdef`'s.